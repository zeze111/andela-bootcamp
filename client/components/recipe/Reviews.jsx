import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import PreLoader from '../common/PreLoader';

/** shows all the reviews for one recipe
 *
 * @class Reviews
 *
 * @extends {React.Component}
 */
class Reviews extends Component {
  /** calls action to delete a user's review
   *
   * @memberof Home
   *
   * @return {void}
   */
  reviewClickEvent = () => {
    this.props.deleteReview(this.props.review.id);
  }

  /** html component to render
   *
   * @memberof Home
   *
   * @return {void}
   */
  render() {
    if (this.props.isLoading) {
      return (
        <div className="center-align loader-style">
          <PreLoader />
        </div>
      );
    }
    const { review, user } = this.props;

    const reviewUser = (
      <div className="secondary-content div-pointer2">
        <i
          role="button"
          tabIndex="0"
          onKeyPress={this.handleKeyPress}
          onClick={this.reviewClickEvent}
          className="material-icons icon-color"
        >delete
        </i>
      </div>
    );

    const nonReviewUser = (
      <span />
    );

    return (
      <li className="collection-item avatar">
        <img src={review.User.image || '/images/profilepic.png'} alt="" className="circle" />
        <span className="review-title "> {review.title} </span>
        <p />
        <p id="r-cmt"> {review.comment} </p>
        <span className="user caps review-list">
          Posted by {review.User.firstName}, {moment(new Date(`${review.createdAt}`)).fromNow()}
        </span>
        {(review.userId === user.id) ? reviewUser : nonReviewUser}
      </li>
    );
  }
}

Reviews.propTypes = {
  deleteReview: PropTypes.func.isRequired,
  review: PropTypes.objectOf(PropTypes.any).isRequired,
  user: PropTypes.objectOf(PropTypes.any),
  isLoading: PropTypes.bool.isRequired
};

Reviews.defaultProps = {
  user: {}
};


export default Reviews;
