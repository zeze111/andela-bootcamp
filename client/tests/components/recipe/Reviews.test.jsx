import React from 'react';

import Reviews from '../../../components/recipe/Reviews';
import store from '../../../store';

const props = {
  deleteReview: jest.fn(() => Promise.resolve()),
  review: {
    id: 1,
    title: 'blah',
    User: {
      id: 20,
      firstName: 'Jane'
    }
  },
  isLoading: false
};

const setup = () => {
  const wrapper = shallow(<Reviews {...props} store={store} />);
  return { wrapper }
}

describe('reviewClickEvent', () => {
  it('should be called on delete a review', () => {
    const { wrapper } = setup();

    wrapper.instance().reviewClickEvent();
  });
});

describe('Test for Reviews', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
