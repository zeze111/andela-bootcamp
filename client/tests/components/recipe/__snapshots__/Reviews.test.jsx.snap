// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Reviews should render component 1`] = `
Object {
  "wrapper": <li
    className="collection-item avatar"
  >
    <img
      alt=""
      className="circle"
      src="/images/profilepic.png"
    />
    <span
      className="review-title "
    >
       
      blah
       
    </span>
    <p />
    <p
      id="r-cmt"
    >
       
       
    </p>
    <span
      className="user caps review-list"
    >
      Posted by 
      Jane
      ,
      Invalid date
    </span>
    <div
      className="secondary-content div-pointer2"
    >
      <i
        className="material-icons icon-color"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        delete
      </i>
    </div>
  </li>,
}
`;
